src/main/java/com/training/springpractice/services/CatalogService.java

```diff
@@ -0,0 +1,35 @@
+package com.training.springpractice.services;
+
+import com.training.springpractice.models.Actor;
+import com.training.springpractice.models.Catalog;
+import com.training.springpractice.models.Movie;
+import com.training.springpractice.models.MovieDerived;
+import com.training.springpractice.repositories.ActorRepository;
+import com.training.springpractice.repositories.CatalogRepository;
+import com.training.springpractice.repositories.MovieDerivedRepository;
+import com.training.springpractice.repositories.MovieRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service("catalogService")
+public class CatalogService {
+
+    @Autowired
+    private CatalogRepository repository;
+
+    @Autowired
+    private MovieRepository movieRepository;
+
+    public CatalogService() {}
+
+    public Catalog create (Catalog catalog) throws Exception {
```

Comment:
Why you didn't need to add @Transactional annotation in this method that is performing an insert in the db?